libraries/abstractions/platform/freertos/iot_network_freertos.c

```diff
@@ -569,6 +573,49 @@ size_t IotNetworkAfr_Receive( void * pConnection,
 
 /*-----------------------------------------------------------*/
 
+size_t IotNetworkAfr_ReceiveUpto( void * pConnection,
+                                  uint8_t * pBuffer,
+                                  size_t bufferSize )
+{
+    int32_t socketStatus = 0;
+    size_t bytesReceived = 0;
+
+    /* Cast network connection to the correct type. */
+    _networkConnection_t * pNetworkConnection = ( _networkConnection_t * ) pConnection;
+
+    /* Caller should never pass a zero-length buffer. */
+    configASSERT( bufferSize > 0 );
+
+    /* Write the buffered byte. THIS IS A TEMPORARY WORKAROUND AND ASSUMES THIS
+     * FUNCTION IS ALWAYS CALLED FROM THE RECEIVE CALLBACK. */
+    if( pNetworkConnection->bufferedByteValid == true )
+    {
+        *pBuffer = pNetworkConnection->bufferedByte;
+        bytesReceived = 1;
+        pNetworkConnection->bufferedByteValid = false;
+    }
+
+    /* Block and wait for incoming data. */
+    socketStatus = SOCKETS_Recv( pNetworkConnection->socket,
+                                 pBuffer + bytesReceived,
+                                 bufferSize - bytesReceived,
```

Comment:
If we call this funciton with 1 here we will end up calling SOCKETS_Recv with 0. Actually why do we have the special logic/workaround above for receiving the first byte?

Reply:
That is because polling is not implemented yet.